qiskit/transpiler/__init__.py

```diff
@@ -155,15 +228,599 @@
+<transpiler-preset-stage-init>`, the input to this stage is a circuit that is already
+ISA-compatible, so a low-level optimization plugin can be tailored for a particular ISA.
+
+There are very few requirements on an optimization plugin, other than it takes in ISA-supported
+circuits, and returns ISA-supported circuits.  An optimization plugin will often contain a loop,
+such as the :class:`.DoWhileController`, and may choose to include the configured translation stage
+as a fix-up pipeline.
+
+Qiskit's built-in optimization plugins are general, and apply well to most real-world ISAs for
+non-error-corrected devices.  The built-in plugins are less well-suited to ISAs that have no
+continuously parametrized single-qubit gate, such as a Clifford+T basis set.
+
+When writing :ref:`stage plugins <transpiler-preset-stage-plugins>`, the entry point for
+``optimization`` is ``qiskit.transpiler.optimization``.  The built-in plugins are:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Summary
+
+    * - :ref:`default <transpiler-preset-stage-optimization-default>`
+      - A default set of optimization passes.  This varies significantly between optimization
+        levels.
+
+.. _transpiler-preset-stage-optimization-default:
+
+Built-in ``default`` plugin
+...........................
+
+This varies significantly between optimization levels.
+
+The specifics of this pipeline are subject to change between Qiskit versions. The broad principles
+are described below.
+
+At optimization level 0, the stage is empty.
+
+At optimization level 1, the stage does matrix-based resynthesis of runs of 1q gates, and very
+simply symbolic inverse cancellation of two-qubit gates, if they appear consecutively.  This runs
+in a loop until the size and depth of the circuit are fixed.
+
+At optimization level 2, in addition the optimizations of level 1, the loop contains commutation
+analysis of sets of gates to widen the range of gates that can be considered for cancellation.
+Before the loop, runs of both one- and two-qubit gates undergo a single matrix-based resynthesis.
+
+At optimization level 3, the two-qubit matrix-based resynthesis runs inside the optimization loop.
+The optimization loop condition also tries multiple runs and chooses the minimum point in the case
+of fluctuating output; this is necessary because matrix-based resynthesis is relatively unstable in
+terms of concrete gates.
+
+Optimization level 3 is typically very expensive for large circuits.
+
+
+.. _transpiler-preset-stage-scheduling:
+
+Scheduling stage
+----------------
+
+.. seealso::
+    :ref:`transpiler-scheduling-description`
+        A guide-level explanation of scheduling concepts.
+
+The scheduling stage, if requested, is responsible for inserting explicit :class:`~.circuit.Delay`
+instructions to make idle periods of qubits explicit.  Plugins may optionally choose to do
+walltime-sensitive transformations, such as inserting dynamical decoupling sequences.
```

Comment:
```suggestion
wall clock time-sensitive transformations, such as inserting dynamical decoupling sequences.
```

Reply:
"walltime" is an equally common spelling of this, and I think we'd end up with "wall-clock-time-sensitive" if we did that, which seems a bit verbose